core/server/services/mega/mega.js

```diff
@@ -168,15 +168,21 @@ async function listener(emailModel, options) {
         return;
     }
 
-    sendEmail(post, members)
-        .then(async () => {
-            return models.Email.edit({
-                status: 'sent'
-            }, {
-                id: emailModel.id,
-                context: {internal: true}
-            });
-        });
+    await models.Email.edit({
+        status: 'submitting'
+    }, {
+        id: emailModel.id,
+        internalContext
```

Comment:
@gargol 🤔 This needs to be `context: internalContext.context` as its not being used like a param here like rest of the places in this file so will just end up expanding itself as the key `internalContext` -
```
internalContext: {context: {internal: true}}
````

Reply:
Yeah, you are :100: right on this! No need to break convention here for no reason :+1: